CMakeLists.txt

```diff
@@ -23,50 +31,61 @@ include(CheckCXXCompilerFlag)
 include(AddCXXCompilerFlag)
 include(CXXFeatureCheck)
 
-# Try and enable C++11. Don't use C++14 because it doesn't work in some
-# configurations.
-add_cxx_compiler_flag(-std=c++11)
-if (NOT HAVE_CXX_FLAG_STD_CXX11)
-  add_cxx_compiler_flag(-std=c++0x)
+if (NOT "${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+  # Try and enable C++11. Don't use C++14 because it doesn't work in some
+  # configurations.
+  add_cxx_compiler_flag(-std=c++11)
+  if (NOT HAVE_CXX_FLAG_STD_CXX11)
+    add_cxx_compiler_flag(-std=c++0x)
+  endif()
 endif()
 
 # Turn compiler warnings up to 11
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
```

Comment:
can we start with making the compiler ID check the outer conditional and bringing all these sections together:

```
if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
  add_cxx_compiler_flag(-W4)
  add_definitions(-D_CRT_SECURE_NO_WARNINGS)
  if (BENCHMARK_ENABLE_LTO)
    set(CMAKE_CXX_FLAGS_RELEASE "/GL")
    ....
  endif()
else()
  add_cxx_compiler_flag(-Wall)
  add_cxx_compiler_flag(-Wextra)
  add_cxx_compiler_flag(-Wshadow)
  ...
  if (BENCHMARK_ENABLE_LTO)
    add_cxx_compiler_flag(-flto)
    ...
  endif()
endif()
```